### ISSUE TITLE
Missing @ReactModule annotation, getNativeModule fails on Android

### ISSUE THREAD

**Issue by minosvasilias** [dooboolab/react-native-iap#1884]
**Description**

I am looking into using this module in an app that requires both react-native as well as native Android/iOS code to access the IAP functionality. On the Android side, the native part of the app is added via `.aar` library. 

When compiling that library, i use a `compileOnly` implementation of this module (specifically [RNIapModule](https://github.com/dooboolab/react-native-iap/blob/main/android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt#L38)), since there is no point double-stacking the dependency. `react-native-iap` will already be a dependency on the react-native side of the app, so the `.aar` library only needs to know about it at compile-time.

I have been successfully using this approach in production with other RN-modules such as [async-storage](https://github.com/react-native-async-storage/async-storage).

Compiling also succeeds here, however when attempting to access `RNIapModule` at runtime, i am faced with this error:
```
java.lang.IllegalArgumentException: Could not find @ReactModule annotation in com.dooboolab.RNIap.RNIapModule
``` 
This happens when calling `reactApplicationContext.getNativeModule(RNIapModule.class)` from the native library.

Above mentioned [AsyncStorageModule](https://github.com/react-native-async-storage/async-storage/blob/36102818f5b6aaff84bb25e248b8ceb89acc7735/android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStorageModule.java#L35) (and many other RN modules) implement this `@ReactModule` annotation, however [RNIapModule](https://github.com/dooboolab/react-native-iap/blob/413ede987f8237a4074076dbecd6e3461a3ae0f5/android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt#L38) does not.
Is there a specific reason this is not the case here? If not, could it be added to support this `getNativeModule` usecase?

**Expected Behavior**

Successful [getNativeModule](https://github.com/facebook/react-native/blob/f3def13a92f0fce72a76d9fc5bdcf619e94688af/ReactAndroid/src/main/java/com/facebook/react/bridge/ReactContext.java#L179) call.

**Environment:**

- react-native-iap: 10.0.0
- react-native: 0.66
- Platforms (iOS, Android, emulator, simulator, device): Android

**Comment by andresesfm** [comment]
Hello @minosvasilias I can see they added that annotation here: https://github.com/react-native-async-storage/async-storage/commit/fbaef6cf72fd53eab7d46a6035aeb4631d43fa39
and here: https://github.com/facebook/react-native/commit/3d1b79cd156e8055eab54cefa84a3532572be82c#diff-3ed0194b1ea14b9f11455ae371a7bc901ec028ca1622b6237eb0f31ef3b351eb

However, I don't see it in the documentation here: https://reactnative.dev/docs/native-modules-android
Also, it's not clear to me from the class documentation: https://github.com/facebook/react-native/blob/main/ReactAndroid/src/main/java/com/facebook/react/module/annotations/ReactModule.java

Could you please point us to some documentation? I see you have a specific use case but I want to get more context as to what the benefits/drawbacks of adding it are.

**Comment by minosvasilias** [comment]
Hey @andresesfm, thanks for the quick reply, appreciate it!

You're right that this is not mentioned in the docs (though it probably should be), however we can follow the `getNativeModule` method in question, which is implemented [here](https://github.com/facebook/react-native/blob/f3def13a92f0fce72a76d9fc5bdcf619e94688af/ReactAndroid/src/main/java/com/facebook/react/bridge/ReactContext.java#L179). 

It is returning the result of `mCatalystInstance.getNativeModule(nativeModuleInterface)`, which is implemented [here](https://github.com/facebook/react-native/blob/f3def13a92f0fce72a76d9fc5bdcf619e94688af/ReactAndroid/src/main/java/com/facebook/react/bridge/CatalystInstanceImpl.java#L468).
That then is retrieving the module name via [getNameFromAnnotation](https://github.com/facebook/react-native/blob/f3def13a92f0fce72a76d9fc5bdcf619e94688af/ReactAndroid/src/main/java/com/facebook/react/bridge/CatalystInstanceImpl.java#L497), which is also where the error i am getting is being thrown.
Meaning that without a valid `ReactModule` annotation, this `ReactContext.getNativeModule` call will always fail.

Regarding drawbacks, i do not believe there are any. `ReactContext.getNativeModule` is however quite a critical method when attempting to access native modules from native code, so i'd argue that it is good practice to implement this for any native module.

**Comment by andresesfm** [comment]
Yeah, I don't see a problem with adding it. Implementation here: https://github.com/dooboolab/react-native-iap/pull/1886

**Comment by andresesfm** [comment]
Released in v10.0.1. Please consider supporting us. Any contribution is greatly appreciated

**Comment by minosvasilias** [comment]
Wonderful, thank you very much! :)
